var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -64,28 +68,30 @@ class Hdf5(AutotoolsPackage):
 
     variant('mpi', default=True, description='Enable MPI support')
     variant('szip', default=False, description='Enable szip support')
+    variant('zlib', default=True, description='Enable zlib support')
     variant('pic', default=True,
             description='Produce position-independent code (for shared libs)')
-    # Build HDF5 with API compaitibility.
-    variant('api', default='none', description='choose api compatibility', values=('v114', 'v112', 'v110', 'v18', 'v16'), multi=False)
+    # Build HDF5 with API compatibility.
+    variant('api', default='none', description='Choose api compatibility', values=('v114', 'v112', 'v110', 'v18', 'v16'), multi=False)
 
     conflicts('api=v114', when='@1.6:1.12.99', msg='v114 is not compatible with this release')
     conflicts('api=v112', when='@1.6:1.10.99', msg='v112 is not compatible with this release')
     conflicts('api=v110', when='@1.6:1.8.99', msg='v110 is not compatible with this release')
     conflicts('api=v18', when='@1.6:1.6.99', msg='v18 is not compatible with this release')
 
-    depends_on('autoconf', type='build', when='@develop')
-    depends_on('automake', type='build', when='@develop')
-    depends_on('libtool',  type='build', when='@develop')
-    depends_on('m4',       type='build', when='@develop')
+    depends_on('cmake@3.12.4:', type='build')
 
     depends_on('mpi', when='+mpi')
-    depends_on('java', when='+java')
+    depends_on('java', type=('build','run'), when='+java+shared')
```

Comment:
What happens when we build `+java~shared`? I would simply put `when='+java'` and introduce an additional conflict `conflicts('+java', when='~shared')`.

Reply:
Done.